string_grouper/string_grouper.py

```diff
@@ -651,6 +617,22 @@ def _validate_replace_na_and_drop(self):
                 "index if the number of index-levels does not equal the number of index-columns."
             )
 
+    @staticmethod
+    def _symmetrize_matrix(AA: csr_matrix) -> csr_matrix:
+        A = AA.tolil()
+        r, c = A.nonzero()
+        A[c, r] = A[r, c]
+        return A.tocsr()
+
+    @staticmethod
+    def _get_matches_list(matches: csr_matrix) -> pd.DataFrame:
```

Comment:
Awesome, so much simpler now... Makes me think what the hell I was thinking when I first wrote that..

Reply:
From my own experience, the optimized version usually doesn't come to mind on the first try.  :)